src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -60,69 +60,57 @@ static void client_channel_factory_unref(
 static grpc_subchannel_args *get_secure_naming_subchannel_args(
     grpc_exec_ctx *exec_ctx, const grpc_subchannel_args *args) {
   grpc_channel_credentials *channel_credentials =
-      grpc_find_channel_credentials_in_args(args->args);
-  if (channel_credentials == NULL) return NULL;
-
+      grpc_channel_credentials_find_in_args(args->args);
+  // Make sure security connector does not already exist in args.
+  if (grpc_security_connector_find_in_args(args->args) != NULL) {
+    gpr_log(GPR_ERROR, "Security connector already present in channel args.");
+    return NULL;
+  }
+  if (channel_credentials == NULL) {
```

Comment:
Suggest moving this check up to line 64, right after we get channel_credentials from the channel args.